### PR TITLE
Warn on FreeType missing glyphs.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Redo of #9175, which was partially reverted in #11615, but with an additional warnings filter to fix the warnings spamming that led to the reversion.
See #12597 (and #8765) for cases where the warning helps (feel free to push a better wording for the warning, e.g. point people to set the correct font).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: src/ft2font.cpp [hunk 1]
```diff
@@ -162,9 +162,19 @@ FT2Image::draw_rect_filled(unsigned long x0, unsigned long y0, unsigned long x1,
 
 inline double conv(long v)
 {
-    return double(v) / 64.0;
+    return v / 64.;
 }
 
+FT_UInt ft_get_char_index_or_warn(FT_Face face, FT_ULong charcode)
```
Comment (QuLogic) [inline src/ft2font.cpp]: Mark `static`?

Comment (anntzer) [reply src/ft2font.cpp]: done

File: src/ft2font.cpp [hunk 2]
```diff
@@ -162,9 +162,19 @@ FT2Image::draw_rect_filled(unsigned long x0, unsigned long y0, unsigned long x1,
...
 {
-    return double(v) / 64.0;
+    return v / 64.;
 }
 
+static FT_UInt ft_get_char_index_or_warn(FT_Face face, FT_ULong charcode)
+{
+    FT_UInt glyph_index = FT_Get_Char_Index(face, charcode);
+    if (!glyph_index) {
+        PyErr_WarnEx(NULL, "Required glyph missing from current font.", 1);
```
Comment (QuLogic) [inline src/ft2font.cpp]: Add `charcode` to message? Also, this might raise; I don't know what you'll need to do to handle that here.

Comment (QuLogic) [reply src/ft2font.cpp]: What happens when you run with `-Werror` though?

Comment (anntzer) [reply src/ft2font.cpp]: Good catch, that gets a `SystemError: <built-in method set_text of matplotlib.ft2font.FT2Font object at 0x7fe670554100> returned a result with an error set`.
Fortunately there's already machinery to handle that (throwing py::exception{} which is handled by CALL_CPP_FULL).  However it (empirically) looks like PyErr_WarnFormat always returns 0 even if it sets the error?  So I explicitly checked for PyErr_Occurred instead, which is fine, but perhaps you have a better idea of why checking the return value of PyErr_WarnFormat doesn't work?

**Pull request merged**